### PR TITLE
authN-authZ: add oauth2-proxy support for authentication and authorization together with GMC

### PR DESCRIPTION
**Pull request by Ruoyu-y**
## Description

Support GUI-based authentication and authorization with the help of Istio, oauth2-proxy and OIDC provider. 

## Issues

#169 

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [*] New feature (non-breaking change which adds new functionality)

## Dependencies

oauth2-proxy

## Tests

`n/a`

### CODE REVIEW

**Comment by Ruoyu-y** [comment]
Please do NOT merge before #260 got merged

File: microservices-connector/config/authN-authZ/README.md
Comment (irisdingbj) [inline microservices-connector/config/authN-authZ/README.md:243]: Does it matter if we have istio sidecar injeciton for oauth2-proxy and keycloak or not?  better to have this info in the steps

Comment (Ruoyu-y) [reply microservices-connector/config/authN-authZ/README.md:243]: Thanks for the notice. Let me add that

File: microservices-connector/config/authN-authZ/README.md
Comment (zhlsunshine) [inline microservices-connector/config/authN-authZ/README.md:210]: Hi @Ruoyu-y, we may need to tell user how to obtain this `INGRESS_PORT`,  by executing shell command or a link of Istio?

Comment (zhlsunshine) [reply microservices-connector/config/authN-authZ/README.md:210]: The same to `INGRESS_HOST`

File: microservices-connector/config/authN-authZ/README.md
Comment (zhlsunshine) [inline microservices-connector/config/authN-authZ/README.md:228]:
```suggestion
6. Create a new Client Scope with the name `groups` in Keycloak with `Include in Token Scope` set as `On`. Include a mapper of type `Group Membership` and set the `Token Claim Name` to `groups`. If the "Full group path" option is selected, you need to include a "/" separator in the group names defined in the --allowed-group option of OAuth2 Proxy. Example: "/groupname". After creating the Client Scope named `groups` you will need to attach it to your client. Go to Clients and find `<your client's id> -> Client scopes` and add client scope and select `groups` and choose `Optional` and you should now have a client that maps group memberships into the JWT tokens so that Oauth2 Proxy may evaluate them.
```

File: microservices-connector/config/authN-authZ/README.md
Comment (zhlsunshine) [inline microservices-connector/config/authN-authZ/README.md:234]:
```suggestion
7. Create two groups `user` and `viewer` by navigating to Groups -> Create group. Assign role `user` to group `user` and role `viewer` to group `viewer` and add user `mary` as a member of group `user` and `bob` as a member of group `viewer`.
```

File: microservices-connector/config/authN-authZ/README.md
Comment (Ruoyu-y) [reply microservices-connector/config/authN-authZ/README.md]: Done

File: microservices-connector/config/authN-authZ/README.md
Comment (Ruoyu-y) [reply microservices-connector/config/authN-authZ/README.md]: Done

File: microservices-connector/config/authN-authZ/README.md
Comment (Ruoyu-y) [reply microservices-connector/config/authN-authZ/README.md]: Done

**Comment by irisdingbj** [comment]
@Ruoyu-y plz remove the draft status from this PR so that we can get it merged

**Review by zhlsunshine — APPROVED** [review]
Thanks @Ruoyu-y, LGTM!

**Pull request merged**
